### ISSUE TITLE
Python 'regex' module support

### ISSUE THREAD

**Issue by OnlineCop** [firasdib/Regex101#1405]
This may be a duplicate of #1219.

I see multiple issues being opened regarding Python syntax: #1271, #1403, #1404. 

Should we offer users who choose Python as the flavor, a choice to select either the `re` or the `regex` modules?

`regex` is backward-compatible with `re` but extends many of the PCRE features, like inline flags, `\K`, recursion, etc.

**Comment by firasdib** [comment]
Wouldn't their need be filled by simply using the PCRE flavor?

**Comment by OnlineCop** [comment]
Mostly, yes, but if `PCRE` were selected as the flavor, the Code Generator is using the `re` module instead of the `regex` module when you change to the Python language.

Most of those advanced features are supported under the `regex` module, so we might want to change `re` for `regex` when the Flavor is PCRE and the Code Generator language is set to Python.

**Comment by Doqnach** [comment]
Possible alternative is to have multiple code generators for Python? (2 vs 3, re vs regex)

**Comment by genericusername3** [comment]
> Wouldn't their need be filled by simply using the PCRE flavor?

No, actually. The `regex` module supports variable-width lookbehinds, for instance (like ECMAScript, but the named group structure in ECMA is `(?<name>pattern)` instead of `(?P<name>patter)`)

**Comment by firasdib** [comment]
Going forward, regex101 will support the Python 3.14 regex engine.

**Comment by genericusername3** [comment]
I don't think the Python 3.14 regex engine (still the built-in `re` module, I assume) is comparable to the third-party module [`regex`](https://pypi.org/project/regex), so I think for correctness this issue should either remain open or be closed with the same rationale as #440 (also fine)

<small>(Or I'm misunderstanding and you do mean `regex`, in which case please disregard this)</small>

**Comment by firasdib** [comment]
Apologies @genericusername3, I was too quick to close the issue. You're right, this is a different module. I'll reopen.
